modules/ui/src/main/scala/lucuma/ui/aladin/ReactAladin.scala

```diff
@@ -0,0 +1,109 @@
+
+  def onPositionChangedCB(cb: PositionChanged => Callback): Callback =
+    Callback(
+      a.on("positionChanged", (o: JsPositionChanged) => cb(PositionChanged.fromJs(o)).runNow())
+    )
+
+  def onZoomCB(cb: Fov => Callback): Callback =
+    Callback(a.on("zoomChanged", (_: Double) => cb(fov).runNow()))
+
+  def onZoomCB(cb: => Callback): Callback =
+    Callback(a.on("zoomChanged", (_: Double) => cb.runNow()))
+
+  def onFullScreenToggleCB(cb: Boolean => Callback): Callback =
+    Callback(a.on("fullScreenToggled", (t: Boolean) => cb(t).runNow()))
+
+  def onFullScreenToggleCB(cb: => Callback): Callback =
+    Callback(a.on("fullScreenToggled", (_: Boolean) => cb.runNow()))
+
+  def onMouseMoveCB(cb: MouseMoved => Callback): Callback =
+    Callback(a.on("mouseMove", (t: JsMouseMoved) => cb(MouseMoved.fromJs(t)).runNow()))
+
+  def pixelScale: PixelScale =
+    PixelScale(a.getSize()(0) / a.getFov()(0), a.getSize()(1) / a.getFov()(1))
+
+  def increaseZoomCB: Callback =
+    Callback(a.increaseZoom())
+
+  def decreaseZoomCB: Callback =
+    Callback(a.decreaseZoom())
+
+  def fixLayoutDimensionsCB: Callback =
+    Callback(a.fixLayoutDimensions())
+
+  def recalculateViewCB: Callback =
+    Callback(a.recalculateView())
+
+  def requestRedrawCB: Callback =
+    Callback(a.requestRedraw())
+
+  def gotoRaDecCB(c: Coordinates): Callback =
+    Callback(a.gotoRaDec(c.ra.toAngle.toDoubleDegrees, c.dec.toAngle.toSignedDoubleDegrees))
+
+case class ReactAladin(
+  clazz:     Css = Css.Empty,
+  options:   AladinOptions = AladinOptions.Default,
+  target:    js.UndefOr[String] = js.undefined,
+  customize: js.UndefOr[Aladin => Callback] = js.undefined,
+  modifiers: Seq[TagMod] = Seq.empty
+) extends ReactFnProps(ReactAladin.component):
+  inline def addModifiers(modifiers: Seq[TagMod]) = copy(modifiers = this.modifiers ++ modifiers)
+  inline def withMods(mods:          TagMod*)     = addModifiers(mods)
+  inline def apply(mods:             TagMod*)     = addModifiers(mods)
+
+object ReactAladin:
+
+  private type Props = ReactAladin
+
+  def resetAladin(
+    r:     Ref.Full[TopNode, TopNode, html.Div],
+    state: UseState[Option[Aladin]],
+    props: Props,
+    force: Boolean
+  ): Callback =
+    r.get.flatMap {
+      case Some(e) if force || state.value.isEmpty =>
+        CallbackTo(new JsAladin(e, props.options)).flatMap { a =>
+          state.setState(Some(a)) *>
+            props.customize.fold(Callback.empty)(f => f(a))
+        }
+      case _                                       => Callback.empty
+    }
+
+  val component =
+    ScalaFnComponent[Props]: props =>
+      for {
+        state <- useState(none[Aladin])
+        r     <- useRefToVdom[html.Div]
+        _     <- useLayoutEffectWithDeps(props.clazz) { _ =>
+                   state.setState(none) *> resetAladin(r, state, props, true)
+                 }
+        _     <- useLayoutEffectWithDeps(r) { r =>
```

Comment:
I'm a bit surprised by this dependency. It's on a `Ref` but not its value. Does this ever change? Would it be equivalent to put it in a `useLayoutEffectOnMount`?

Reply:
It works with `useLayoutEffectOnMount` which is a bit surprising.
Basically I want to setup the js aladin once the ref is set

Reply:
The ref is set because onMount effects run after the first render.